rasa/core/policies/intent_ted_policy.py

```diff
@@ -482,6 +484,33 @@ def predict_action_probabilities(
             action_metadata=self._collect_action_metadata(domain, similarities),
         )
 
+    @staticmethod
+    def _is_queried_for_first_time(tracker: DialogueStateTracker) -> bool:
```

Comment:
I find the name a bit unintuitive. Maybe `_does_not_end_with_action`

Reply:
Why can't we just do one loop that goes backwards in the applied events and returns `False` once it encounters an `ActionExecuted` and `True` when it finds a `UserUttered`?

Reply:
Regarding the name -> I would have agreed on the name if this function was inside `DialogueStateTracker` class. If it is inside `IntentTEDPolicy` it should describe an activity of `IntentTEDPolicy`. From that perspective, `_is_queried_for_first_time` made sense to me because it checks if `IntentTEDPolicy` is queried for the first time after the last UserUttered.
@kedz @JEM-Mosig Wdyt?

Reply:
What about `_is_queried_before_action_executed`?

Reply:
Perfect! Renamed and refactored the logic.